source/core.py

```diff
@@ -580,16 +580,10 @@ def _doPostNvdaStartupAction():
 
 	queueHandler.queueFunction(queueHandler.eventQueue, _doPostNvdaStartupAction)
 
-
 	log.debug("entering wx application main loop")
 	app.MainLoop()
 
 	log.info("Exiting")
-	if updateCheck:
-		_terminate(updateCheck)
-
-	_terminate(watchdog)
-	_terminate(globalPluginHandler, name="global plugin handler")

```

Comment:
Can you explain why these needed to get moved into gui? I'd really prefer that they stay in core... but perhaps there is a necessary reason why this is not possible...

Reply:
All these things used to run before closing the windows and terminating the gui in 2020.4. It could be worth registering them to an exit event in `source\core.py`, so they aren't lost, and then firing the event in `safeAppExit`.

Reply:
I've moved this to an extension point with comments to justify